### PR TITLE
Support ASP.NET Core 3 (WIP)

### PR DESCRIPTION
**Pull request by 304NotModified**
to-do

- [ ] fix build errors
  -  [ ]  `services.AddHttpContextAccessor();` not found
   - [ ] ` builder.ConfigureServices((builderContext, services) => AddNLogLoggerProvider(services, builderContext.Configuration, options, CreateNLogLoggerProvider));` - Delegate 'Action<IServiceCollection>' does not take 2 arguments	           `
- [ ] unit tests
- [ ] check `#if
- [ ] `Microsoft.AspNetCore.Http.HttpRequestRewindExtensions.EnableBuffering(request)` is gone?

### CODE REVIEW

File: src/NLog.Web.AspNetCore/NLog.Web.AspNetCore.csproj [hunk 1]
```diff
@@ -73,6 +77,10 @@ Supported platforms:
     <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.1.1" />
     <PackageReference Include="Microsoft.AspNetCore.Routing.Abstractions" Version="2.1.1" />
   </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.0' ">
+    <PackageReference Include="Microsoft.Extensions.Http" Version="3.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="3.0.0" />
```
Comment (304NotModified) [inline src/NLog.Web.AspNetCore/NLog.Web.AspNetCore.csproj]: @snakefoot this is a good reason to support the new stuff,  the packages are renamed :|

**Comment by 304NotModified** [comment]
I'm done for today but someone knows how to fix the build issues, please let me know. (or send a commit/PR).

**Comment by 304NotModified** [comment]
Todo check https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.0&tabs=visual-studio

**Comment by 304NotModified** [comment]
Check consequence AllowSynchronousIO disabled

**Comment by 304NotModified** [comment]
Double check HostBuilder replaces WebHostBuilder

**Comment by snakefoot** [comment]
According to this:

https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.0&tabs=visual-studio#migrate-libraries-via-multi-targeting

Then the dependency should be like this:

```xml
  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.0' ">
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>
```

**Comment by snakefoot** [comment]
Created PR with a green build: #496

**Comment by 304NotModified** [comment]
superseded by #496

**Pull request closed without merging**